openvdb/tools/PointPartitioner.h

```diff
@@ -918,11 +878,49 @@ inline void partition(
     PointIndexType* index = pointIndices.get();
     for (size_t n = 0; n < numSegments; ++n) {
         indexArray.push_back(index);
-        index += offestSegments[n]->size();
+        index += offsetSegments[n]->size();
     }
 
+    // compute leaf node origin for each page
+
+    pageCoordinates.reset(new Coord[pageCount]);
+
+    tbb::parallel_for(segmentRange,
+        [&](tbb::blocked_range<size_t>& range)
+        {
+            for (size_t n = range.begin(); n < range.end(); n++)
+            {
+                Index segmentOffset = segmentOffsets[n];
+                PointIndexType* indices = pageIndexArrays[n].get();
+
+                const Coord& segmentCoord = segmentCoords[n];
+
+                tbb::blocked_range<size_t> copyRange(0, pageOffsetArrays[n][0] - 1);
+                tbb::parallel_for(copyRange,
+                    [&](tbb::blocked_range<size_t>& r)
+                    {
+                        for (size_t i = r.begin(); i < r.end(); i++)
+                        {
+                            Index pageIndex = indices[i];
+                            Coord& ijk = pageCoordinates[segmentOffset+i];
+
+                            ijk[0] = pageIndex >> (2 * binLog2Dim);
+                            Index pageIndexModulo = pageIndex - (ijk[0] << (2 * binLog2Dim));
+                            ijk[1] = pageIndexModulo >> binLog2Dim;
+                            ijk[2] = pageIndexModulo - (ijk[1] << binLog2Dim);
+
+                            ijk = (ijk << bucketLog2Dim) + segmentCoord;
```

Comment:
Can you elaborate on how this logic maps to the original behaviour of `LeafNodeOriginOp`

Reply:
The original behavior of `LeafNodeOriginOp` re-computed the leaf node by re-applying the binning operation to the first point in the bin. Instead, this algorithm stores the page indices computed in OrderSegmentsOp (which would otherwise be thrown away) and then uses the inverse of the index portion of the algorithm in BinPointIndicesOp to deduce the leaf node origin from this stored index.

PS I'm really not fan of the terminology in this tool - in particular **buckets** means the same as **pages** and **leafs**, **bins** means the same as **segments** and the frequent interchange of these terms makes the algorithm quite hard to follow IMO. Nevertheless, I've attempted to follow existing conventions in this isolated change. There's more refactoring to come to try and make the algorithm as a whole more readable.

Reply:
Okay great, thanks - I haven't verified the inverse calculation here but your explanation sounds good.

Reply:
@danrbailey Given that you found the terminology confusing I suggest you do one of two - either document it (the easiest) or attempt to fix it (I assume this is a tall order). Anyway, confusing terminology in code is a really bad thing and should be addressed in some way (it makes the code hard to maintain).

Reply:
Thanks @kmuseth! I will merge this PR as-is and then follow up with a new PR that refactors portions of this algorithm to fix the terminology as well as introduce some C++11 improvements that together will hopefully improve maintainability.